src/logic/safe/store/reducer/allTransactions.ts

```diff
@@ -0,0 +1,34 @@
+import { handleActions } from 'redux-actions'
+
+import { Transaction } from '../models/types/transactions'
+import { LOAD_MORE_TRANSACTIONS } from '../actions/allTransactions/pagination'
+
+export const TRANSACTIONS = 'allTransactions'
+
+export interface TransactionsState {
+  [safeAddress: string]: Transaction[]
+}
+
+export default handleActions(
+  {
+    // todo: because we are thinking in remove immutableJS, I will implement this without it so it can be easier removed in future
+    [LOAD_MORE_TRANSACTIONS]: (
+      state: TransactionsState,
+      action: {
```

Comment:
The type for the action should be defined alongside with the action itself so it's reusable in the future

Reply:
I'm not sure if I understood you, could you please clarify?

Reply:
The type for the action should be defined close to the action and not in the reducer. How would you reuse the type for this action?